pkg/webhook/resourcedistribution/validating/utils.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2021 The Kruise Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validating
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+const (
+	DefaultNamespace                     = "default"
+	ResourceHashCodeAnnotation           = "kruise.io/resourcedistribution.resource.hashcode"
+	ResourceDistributedTimestamp         = "kruise.io/resourcedistribution.resource.distributed.timestamp"
+	SourceResourceDistributionOfResource = "kruise.io/resourcedistribution.resource.from"
+)
+
+var (
+	// supportedGVKList is a list that contains all supported resource group, version, and kind
+	// Support CustomResourceDefinition
+	/* ADD NEW RESOURCE TYPE HERE*/
+	supportedGVKList = []schema.GroupVersionKind{
+		{Group: "", Version: "v1", Kind: "Secret"},
+		{Group: "", Version: "v1", Kind: "ConfigMap"},
+	}
+
+	// ForbiddenNamespaces is a list that contains all forbidden namespaces
+	// Resources will never be distributed to these namespaces
+	// reused by controller
+	/* ADD NEW FORBIDDEN NAMESPACE HERE*/
+	ForbiddenNamespaces = []string{
+		"kube-system",
+		"kube-public",
+	}
+)
+
+// isSupportedGVK check whether object is supported by ResourceDistribution
+func isSupportedGVK(object runtime.Object) bool {
+	if object == nil {
+		return false
+	}
+	objGVK := object.GetObjectKind().GroupVersionKind()
```

Comment:
Should judge only group and kind?

Reply:
Sounds reasonable. @furykerry @FillZpp  what do you think?